runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -120,6 +138,20 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
     @Override
     public <R> R access(ICorfuSMRAccess<R, T> accessMethod,
                         Object[] conflictObject) {
+        R value;
+        boolean isEnabled = MetricsUtils.isMetricsCollectionEnabled();
```

Comment:
Once again, I don't understand why obtain the isEnabled boolean from MetricUtils, only to send it back to MetricUtils.getConditionalContext(). Same below in all instances.